docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,243 @@
+
+- **Inside the body of the context manager (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), retrieve a log and report an asset materialization.** These calls use the temporary communications channels established by <PyObject object="PipesDefaultContextLoader" module="dagster_pipes" /> and <PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />. To see the full range of what you can do with the <PyObject object="PipesContext" module="dagster_pipes" />, see the API docs or the general [Pipes documentation](/guides/dagster-pipes).
+
+At this point you can execute the rest of your Kubernetes code as normal, invoking various <PyObject object="PipesContext" module="dagster_pipes" /> APIs as needed.
+
+#### Existing codebases
+
+For illustrative purposes, we've created a Python script from scratch. However, you may want to apply Pipes to an existing codebase.
+
+One approach that can be useful is to wrap the <PyObject object="open_dagster_pipes" module="dagster_pipes" /> context manager around an existing `main` function or entry point. You can either pass the <PyObject object="PipesContext" module="dagster_pipes" /> down through your business logic, or simply report an asset materialization after your business logic is done:
+
+```python
+from dagster_pipes import (
+    PipesContextLoader,
+    PipesMessageWriter,
+    open_dagster_pipes,
+)
+
+# ... existing code
+
+if __name__ == "__main__":
+    with open_dagster_pipes(
+        context_loader=PipesContextLoader(),
+        message_writer=PipesMessageWriter(),
+    ) as pipes:
+        # ... existing logic
+        pipes.report_asset_materialization(
+            asset_key="raw_orders",
+            metadata={"some_key": "some_value"},
+            data_version="alpha",
+        )
+```
+
+---
+
+## Step 3: Launch the Kubernetes pod from the Dagster UI
+
+In this step, you’ll run the Kubernetes pod you created in [Step 1.1](#step-11-define-the-dagster-asset) from the Dagster UI.
+
+1. In a new command line session, run the following to start the UI:
+
+   ```python
+   dagster dev -f dagster_k8s_pipes.py
+   ```
+
+2. Navigate to [localhost:3000](http://localhost:3000/), where you should see the UI: \[SCREENSHOT]
+
+3. Click **Materialize** near the top right corner of the page, then click **View** on the **Launched Run** popup. Wait for the run to complete, and the event log should look like this:
+
+   <Image
+   alt="Event log for Databricks run"
+   src="/images/guides/dagster-pipes/databricks/run.png"
+   width={2554}
+   height={1335}
+   />
```

Comment:
TODO - Placeholder; change this to be Kubernetes and not Databricks

@alangenfeld Would you be able to help with getting a screenshot for this?

Reply:
ya I can